src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -294,8 +294,17 @@ export class HubConnection {
             return this._stopPromise!;
         }
 
+        const state = this._connectionState;
         this._connectionState = HubConnectionState.Disconnecting;
 
+        if (state === HubConnectionState.Connected) {
+            await this._sendWithProtocol(this._createCloseMessage());
+            // Transport can close during the above 'await', we should check if that occurred and noop.
```

Comment:
In this scenario, stop has fully completed and `_stopPromise` would already be undefined, right?